servers/rendering/rendering_device.h

```diff
@@ -597,7 +597,10 @@ class RenderingDevice : public Object {
 	/**** SHADER ****/
 	/****************/
 
-	virtual Vector<uint8_t> shader_compile_from_source(ShaderStage p_stage, const String &p_source_code, ShaderLanguage p_language = SHADER_LANGUAGE_GLSL, String *r_error = nullptr, bool p_allow_cache = true);
+	virtual uint32_t get_vulkan_major() const = 0;
+	virtual uint32_t get_vulkan_minor() const = 0;
+
+	virtual Vector<uint8_t> shader_compile_from_source(ShaderStage p_stage, const String &p_source_code, ShaderLanguage p_language = SHADER_LANGUAGE_GLSL, String *r_error = nullptr, bool p_allow_cache = true, uint32_t p_vulkan_major = 1, uint32_t p_vulkan_minor = 0);
```

Comment:
It feels a bit weird to have the Vulkan version passed as argument and re-evaluated each time we compile a shader, sounds like something that should be evaluated at startup and the result cached? Not that it's particularly a performance concern but that's typical "init" stuff IMO that I would expect to only happen when starting the app.

Not blocking for this PR though, just a note.

Reply:
Neither me nor Reduz had any idea how we could make this nice especially seeing there may be more too this. So yeah agree with the remark, don't have a good alternative ready

Reply:
I'm now storing all info in a struct and giving access to that struct. Still passing it as a parameter for now.